lib/Gocdb_Services/APIAuthenticationService.php

```diff
@@ -274,9 +257,103 @@ private function validate($data, $identifier, $type)
         //If the entity is of type OIDC subject, do a more thorough check again
         if (
             $type == 'OIDC Subject' &&
-            !preg_match("/^([a-f0-9]{8}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{12})$/", $identifier)
+            !preg_match(
+                "/^([a-f0-9]{8}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{12})$/",
+                $identifier
+            )
         ) {
             throw new \Exception("Invalid OIDC Subject");
         }
```

Comment:
We have very similar checks on OIDC subject validity here and in `/lib/Gocdb_Services/Site.php` (and of course the checks differ slightly, though I believe this one is more correct.

Factoring this out, maybe pushing it (and X.509 validation as above) into the existing validation layer is a way forward.

@Sae126V, can you open an issue?